Support time inputs

Antsiscool opened the pull request:
I am converting a project over from capybara selenium and found that time inputs were failing to be correctly set.

This allows specifying the time as a string or a Time object. The date attributes will be ignored.

I have added test cases for both string and time. I also added some test cases for date inputs as well.

Antsiscool commented:
I am not sure why the tests are failing. They are all failing for me locally on the main branch as well.

Antsiscool commented:
I have just rebased my changes onto the latest main branch.

route commented:
Just approved them to run

route commented:
Weird, dunno let me release new Ferrum and push my changes to cuprite and I'll get this into new release

janwerkhoven commented:
Need this. +1  👀

Antsiscool commented:
@route I have revisited this PR and cleaned it up. The issue with the original failing tests was that loading SVG icons for time and date inputs and this was increasing the count of network requests.

I have moved the time and date tests to a new HTML file and this has resolved the problem.

I have also added support for month and week inputs.

It seems the test suite is failing on Ruby 3.0 and Ruby 3.2. I have been unable to replicate these failures locally.

Antsiscool commented:
These failing tests seem to be flaky. I can replicate the issue locally.

I created a PR onto my fork to run the tests and they all passed. https://github.com/Antsiscool/cuprite/pull/1

route commented:
Rebasing and running tests locally. If they all pass, I'll merge manually
